### PR TITLE
Use GovukComponent::Details

### PR DESCRIPTION
**Pull request by AgaDufrat**
Tech debt https://trello.com/c/Ci0WZjb2/353-low-add-details-component

### CODE REVIEW

File: app/components/candidate_interface/rejection_reasons_component.html.erb [hunk 1]
```diff
@@ -1,15 +1,10 @@
...
-
-  <div class="govuk-details__text govuk-!-padding-bottom-0">
-    <% rejected_application_choices.each do |application_choice| %>
-      <%= render(SummaryCardComponent.new(rows: application_choice_rows(application_choice))) do %>
-        <%= render(SummaryCardHeaderComponent.new(title: application_choice.provider.name)) %>
-      <% end %>
+<%= render GovukComponent::Details.new(
+  summary: 'See feedback from your previous application',
+),
+  classes: 'govuk-!-padding-bottom-0' do %>
```
Comment (benswannack) [inline app/components/candidate_interface/rejection_reasons_component.html.erb:4]: Do the classes need to go inside the brackets here?

**Review by benswannack — APPROVED** [review]
Nice, minor thing inline

File: app/components/candidate_interface/rejection_reasons_component.html.erb (hunk 1 above)
Comment (AgaDufrat) [reply app/components/candidate_interface/rejection_reasons_component.html.erb:4]: good spot! 😅

**Comment by paulrobertlloyd** [comment]
@AgaDufrat For the other components, I’ve been using the helper methods, which are a bit less verbose. Do you think we can do the same for the details component? For example, instead of this:

```erb
<%= render GovukComponent::Details.new(
  summary: 'Email body',
) do %>
  <%= tag.pre(email.body) %>
<% end %>
```

we can do this:

```erb
<%= govuk_details(
  summary: 'Email body',
) do %>
  <%= tag.pre(email.body) %>
<% end %>
```

**Pull request merged**